File: Ruby/lib/palindrome_number.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
  return false if x < 0

  # @param {Integer} n
  # @return {Integer}
  reverse = ->(n) {
    result = 0

    while n != 0
      last_digit = n % 10
      result = result * 10 + last_digit
      n = (n / 10).floor
    end

    return result
  }

  return reverse.call(x) == x
end

File: Java/src/main/java/PalindromeNumber.java
import java.util.function.Consumer;
import java.util.function.Function;

public class PalindromeNumber {
  public static boolean isPalindrome(int x) {
    if (x < 0) {
      return false;
    }

    Function<Integer, Integer> reverse =
        (num) -> {
          int result = 0;

          while (num != 0) {
            int lastDigit = num % 10;
            result = result * 10 + lastDigit;
            num /= 10;
          }

          return result;
        };

    return x == reverse.apply(x);
  }
}
